torch/nn/modules/transformer.py

```diff
@@ -462,6 +462,14 @@ def forward(
             )
         elif src_key_padding_mask is None:
             why_not_sparsity_fast_path = "src_key_padding_mask was None"
+        # This check avoids a call to torch._nested_tensor_from_mask_left_aligned() that
+        # breaks in torch.compile.
+        elif (
+            (not hasattr(self, "mask_check")) or self.mask_check
```

Comment:
I know this is used in the branch just below, but to confirm, if not hasattr(self, "mask_check"), this means the mask is always checked?

Reply:
yes that's correct.. the default is to do a mask check if for some reason the attribute isn't present. Doing a mask check is also the default if the attribute was unspecified upon construction.